mqlc/mqlc.go

```diff
@@ -1684,18 +1684,111 @@ func (c *compiler) postCompile() {
 			chunk, typ, ref := c.expandListResource(chunk, ref)
 			switch chunk.Id {
 			case "$one", "$all", "$none", "$any":
+				// default fields
 				ref = chunk.Function.Binding
-				chunk := code.Chunk(chunk.Function.Binding)
+				chunk := code.Chunk(ref)
 				typ = types.Type(chunk.Function.Type)
-				c.expandResourceFields(chunk, typ, ref)
-				block.Datapoints = append(block.Datapoints, block.TailRef(ref))
+				expanded := c.expandResourceFields(chunk, typ, ref)
+				// when no defaults are definied or query isn't about a resource, no block was added
+				if expanded {
+					block.Datapoints = append(block.Datapoints, block.TailRef(ref))
+					c.addValueFieldChunks(ref)
+				}
 			default:
 				c.expandResourceFields(chunk, typ, ref)
 			}
 		}
 	}
 }
 
+// addValueFieldChunks takes the value fields of the assessment and adds them to the
+// block forthe default fields
+// This way, the actual data of the assessment automatically shows up in the output
+// of the assessment that failed the assessment
+func (c *compiler) addValueFieldChunks(ref uint64) {
+	var assessmentBlock *llx.Block
+	var whereChunk *llx.Chunk
+
+	// find chunk with where/whereNot function
+	for {
+		chunk := c.Result.CodeV2.Chunk(ref)
+		if chunk.Function == nil {
+			// this is a safe guard and shouldn't happen
+			log.Error().Msg("failed to find where function for assessment")
+			return
+		}
+		if chunk.Function.Args != nil {
+			whereChunk = chunk
+			break
+		}
+		ref = chunk.Function.Binding
+	}
+
+	// find the referenced blcok for the where function
+	for i := len(whereChunk.Function.Args) - 1; i >= 0; i-- {
+		arg := whereChunk.Function.Args[i]
```

Comment:
I'm not sure about which index it actually is. That's why I iterate over the args.

Reply:
yeah i think it's fine for now, since the where calls have arguments on different places depending on their underlying type. but it really shows that we may want to clean that up and harmonize it to avoid having to guess it here :thinking: